frontend/src/App.js

```diff
@@ -1,55 +1,115 @@
-import logo from './logo.svg';
+import React, { Component } from "react";
+import axios from "axios"
 import './App.css';
 
-class Matchup {
-  constructor(team1, team2) {
-    this.team1 = team1
-    this.team2 = team2
+
+async function callApi(endpoint) {
+  try {
+    const response = await axios.get('/api/' + endpoint);
+    return response.data
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+
+function spliceIntoChunks(arr, chunkSize) {
+  const res = [];
+  while (arr.length > 0) {
+      const chunk = arr.splice(0, chunkSize);
+      res.push(chunk);
   }
+  return res;
 }
 
-const matchup1 = new Matchup('USA', 'Mexico')
-const matchup2 = new Matchup('Spain', 'Portugal')
-const matchup3 = new Matchup('Canada', 'Honduras')
 
-function App() {
+// Return an array containg sub-arrays of 6 matches
+function getChunkedGroupStageMatches(matches) {
+  const groupStageMatches = matches.filter(
+    match => match.comp_stage === "Group Stage");
+  const matchupChunks = spliceIntoChunks(groupStageMatches, 6)
+  return matchupChunks
+ }
+
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      matches: [],
+      teams: []
+    }
+    
+  }
+
+  // Make ALL API calls here so that all future munging is in-memory
+  async componentDidMount() {
 
-  const renderMatchupRow = (matchup) => {
+    const matches = await callApi('matches')
+    const teams = await callApi('teams')
```

Comment:
these are the only times we have to `await` now. probably a best practice to keep API calls highly localized